### PR TITLE
Feature/issue 27318

### PR DESCRIPTION
**Pull request by haiderbari**
## Hey, I just made a Pull Request!

Updated the readme docs on how to use Template Example

issue: https://github.com/backstage/backstage/issues/27318

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: docs/features/software-templates/writing-custom-actions.md [hunk 1]
```diff
@@ -141,6 +141,90 @@ Prefer to use `camelCase` over `snake_case` or `kebab-case` for these actions if
...
+Add the example to the `createTemplateAction` under the object property `examples`:
+
+```ts title="With JSON Schema"
+import { examples } from './acmeExample.examples';
+
+export const createNewFileAction = () => {
+  return createTemplateAction<{ contents: string; filename: string }>({
+    id: 'acme:file:create',
+    description: 'Create an Acme file.',
+    examples,
```
Comment (jhaals) [inline docs/features/software-templates/writing-custom-actions.md:196]: Let's inline the example here instead.

File: docs/features/software-templates/writing-custom-actions.md [hunk 2]
```diff
@@ -141,6 +141,90 @@ Prefer to use `camelCase` over `snake_case` or `kebab-case` for these actions if
...
+2. Refer to the TemplateExample to understand the required structure and components.
+3. Fill in the necessary details in the template, such as action name, parameters, and logic.
+4. Integrate the completed template into your software template to enable the custom action.
+
+#### Benefits
+
+- Serves as templates for users to understand how to configure and use the actions.
+- Ensures consistency in custom actions.
+- Simplifies the creation process by providing a clear structure.
+- Helps in maintaining standardization across different templates.
```
Comment (jhaals) [inline docs/features/software-templates/writing-custom-actions.md:160]:
```suggestion
```

I think your initial text is great and explains why this is a great thing to have, let's keep it short and sweet :)

**Review by jhaals — CHANGES_REQUESTED** [review]
Great job with the docs, just some small tweaks :)

File: docs/features/software-templates/writing-custom-actions.md (hunk 1 above)
Comment (haiderbari) [reply docs/features/software-templates/writing-custom-actions.md:196]: Hi @jhaals do you want me to inline the code from 193 to 196?

Example all on one line

return createTemplateAction<{ contents: string; filename: string }>({ id: 'acme:file:create', description: 'Create an Acme file.',

Comment (haiderbari) [reply docs/features/software-templates/writing-custom-actions.md:196]: I understand I think just to confirm this is what you are looking for 

Base on the createTemplateAction example above add the example under the object property examples:

return createTemplateAction<{ contents: string; filename: string }>({id: 'acme:file:create', description: 'Create an Acme file', examples, ...};

**Review by jhaals — CHANGES_REQUESTED** [review]
Looks like there's a formatting issue. Make sure to run `yarn prettier:fix`

**Comment by haiderbari** [comment]
> Looks like there's a formatting issue. Make sure to run `yarn prettier:fix`

I have applied the fix, thanks

**Review by jhaals — APPROVED** [review]

**Pull request merged**